packages/gasket-request/lib/internal.d.ts

```diff
@@ -0,0 +1,8 @@
+import { Gasket } from '@gasket/core';
+import { RequestLike } from './index.js';
+
+export type withGasketRequest = (actionFn: (gasket: Gasket, req: RequestLike, ...args?: []) => Promise<unknown>) =>
+  (gasket: Gasket, req: RequestLike, ...args?: []) => Promise<unknown>
```

Comment:
I'd make sure you're preserving the types of the original function for the wrapped function; as this is currently it'll return a function that resolves to `Promise<unknown>` and has unknown parameter types. Also, shouldn't the inner function receive the normalized `GasketRequest` type?

```suggestion
export function withGasketRequest<Result, Args extends Array<unknown>>(actionFn: (gasket: Gasket, req: GasketRequest, ...args: Args) => Promise<Result>):
  (gasket: Gasket, req: RequestLike, ...args: Args) => Promise<Result>;
```

Reply:
[Playground link](https://www.typescriptlang.org/play/?#code/C4TwDgpgBA4ghgZwNYWFAvFA3gXwFCiSyIrABKEAjgK4QJqa4HjQU13AAyAlihtvjwATCAGMANnABO0AGbUAdqODcA9gqgB3bsAAW8ZKja16AHgoJq44ABooAQSkBzBFAgAPYBAVDXjqXAgpopICqqaCgB8kQAUcMpqCgBiCgBcUDFOJKjpBqR2MpS52eRUJrZQAHTV0i7pji4AlBiRUAAKUqoAttwIEOZ0VsCRjal4UBlZhsDF0wVU6cYcPCh21ZW1CPXOCM3orR3dvf0WQ5F4eKLq9FAARpK6-Np6eUZlHHEIIEqTJfOUdlkqlU6QU1C6twgUjst2k6XoUm4Cice1aWHGEygMmA1CkGlhUkqMkgcGAMSBqkaAG48DhqRdEN9RFB5EoVOo7nBdDFmujMVArgobhT+HBNHAdHcHjFcHZZVAACx2ADk9y5yvp+CAA) that shows this approach and how the implicit typing preserves the original function parameter/return value typings.

Reply:
How did I not know about this playground... thank you!